src/gwt/src/org/rstudio/studio/client/workbench/views/files/Files.java

```diff
@@ -777,13 +838,75 @@ private void showFileInBrowser(FileSystemItem file)
       }
    }
 
+   // generate a filename based on the file type and currentPath_ variable
+   private FileSystemItem getDefaultFileName(TextFileType fileType) 
+   {
+      String defaultExt = fileType.getDefaultExtension();
+      
+      // extension has a '.' at the start, so remove that in the default name
+      String newFileDefaultName = "new_" + defaultExt.toUpperCase().substring(1) + "_file" + defaultExt.toLowerCase();
+      String path = workingPath_.completePath(newFileDefaultName);
+      FileSystemItem newTempFile = FileSystemItem.createFile(path);
+      return newTempFile;
+   }
+   
+   private void touchFile(TextFileType fileType)
+   {
+      // prepare default information about the new file
+      FileSystemItem newTempFile = getDefaultFileName(fileType);
+      
+      // guard for reentrancy
+      if (inputPending_)
+         return;
+      
+      inputPending_ = true;
+
+      // prompt for new file name then execute the operation
+      globalDisplay_.promptForText("Create and Open New File",
```

Comment:
Since you already have the TextFileType, we could add the description of the file type to this dialog (e.g. "Create and Open New R Markdown File", "Create and Open New JSON File", etc.) What do you think?

Reply:
I like that. I'll implement this for the next revision.